src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -566,4 +566,23 @@ private static void validateSingleQueryType(Integer k, Float distance, Float sco
             throw new IllegalArgumentException(String.format("[%s] requires exactly one of k, distance or score to be set", NAME));
         }
     }
+
+    private static void updateQueryStats(Integer k, Float minScore, Float maxDistance, QueryBuilder filter) {
```

Comment:
nit: you can clean this up a bit more to reduce if else. every parameter similar to k or maxDistance we add, will add to the if else statements.

```
validateSingleQuery()
updateQueryStats(k, ...)
updateQueryStats(minScore, ...)
updateQueryStats(maxDistance, ...)
```
```
static <T> void updateQueryStats(T queyParam, QueryBuilder filter, queryCounter, filtercounter) {
if (queyParam != null) {
            queryCounter.increment();
            if (filter != null) {
                filtercounter.increment();
            }
        }
}
```

Reply:
Thanks @shatejas, this suggestion could reduce the function code itself, but looks like we still need extra multiple time calls at same place. Beside the `k` is Integer type, `maxDistance` and `minScore` are Float type, we couldn't directly use parameter `T queyParam` for all of them.

Reply:
@junqiu-lei Why not use Object?

Reply:
@VijayanB I don't see obviously benefit to introduce Object for this internal used function. Besides if we introduce the Object, we anyway need the condition check for which query type and query counter to use somewhere else.

Reply:
@junqiu-lei yeah extra calls is sometimes a good trade off for maintainability. the cost here shouldn't be high but correct me if I am missing something

@VijayanB On the object suggestion, a few thoughts - The method itself is doing quite a few things - parsing, building query builder, validating and adding stats. Its generally an indication to break it down. What we can actually do is start building query builder while parsing and once we build the KNNQueryBuilder we can validate the entire object as a whole and then update the stats for it as a whole, that way you don't need an extra object.

psuedo code to better understand 
```java
KnnQueryBuilder querybuilder = parseAndBuild(XContentparser) //Note there are no validations here
validate(querybuilder); //This throws if any of the validations fail
updateStats(querybuilder); //update stats
```

Having said this, its too much of a refactor so I wouldn't want to club it with this PR and can be punted till absolutely needed

Reply:
Just had offline with @shatejas, I've updated this method to use generic parameter for queryParam

Reply:
Please check my comment: https://github.com/opensearch-project/k-NN/pull/1684#discussion_r1595893366

Reply:
Updated PR on your comment @navneet1v